gettsim/transfers/wohngeld.py

```diff
@@ -365,54 +331,49 @@ def wohngeld_miete_m_bis_2008(
     -------
 
     """
-    immobilie_baujahr = hh_id.replace(immobilie_baujahr_hh)
     # Get yearly cutoff in params which is closest and above the construction year
     # of the property. We assume that the same cutoffs exist for each household
     # size.
-    yearly_cutoffs = sorted(wohngeld_params["max_miete"][1], reverse=True)
-    conditions = [immobilie_baujahr <= cutoff for cutoff in yearly_cutoffs]
-    constr_year_category = np.select(conditions, yearly_cutoffs)
-    max_definierte_hh_größe = max(
-        i for i in wohngeld_params["max_miete"] if isinstance(i, int)
+    params_max_miete = wohngeld_params["max_miete"]
+    selected_bin_index = np.searchsorted(
+        sorted(params_max_miete[1]), immobilie_baujahr_hh, side="left"
     )
-    data = [
-        wohngeld_params["max_miete"][hh_größe][constr_year][ms]
-        if hh_größe <= max_definierte_hh_größe
-        else wohngeld_params["max_miete"][max_definierte_hh_größe][constr_year][ms]
-        + wohngeld_params["max_miete"]["jede_weitere_person"][constr_year][ms]
-        * (hh_größe - max_definierte_hh_größe)
-        for hh_größe, constr_year, ms in zip(
-            haushaltsgröße, constr_year_category, mietstufe
+
+    constr_year = list(params_max_miete[1])[selected_bin_index]
+
+    # Calc maximal considered rent
+    max_definierte_hh_größe = max(i for i in params_max_miete if isinstance(i, int))
```

Comment:
Please make note / issue to do more pre-processing of yaml contents. We don't want to repeat this a gazillion times, also not very robust.